function Muscle {
    SeqBase=`basename $1`; muscle -in ${Seq} -out ${SeqBase%.fas}-muscle.fa
}
cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA
#
Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_Plus_FL.fas
SeqBase=`basename ${Seq}`; muscle -in ${Seq} -out ${SeqBase%.fas}-muscle.fa
#
Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_Plus_L2L1.fas
SeqBase=`basename ${Seq}`; muscle -in ${Seq} -out ${SeqBase%.fas}-muscle.fa
#
# Ref, [A05051_HPV51REF_M62877_Primer-Guide.gb](/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/A05051_HPV51REF_M62877_Primer-Guide.gb)

Region=0874-7821
Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_Plus_FL-muscle.fa
extractalign ${Input} ${Input%.*}_${Region}.fas -regions ${Region}
#
Region=0001-0874
Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_Plus_FL-muscle.fa
extractalign ${Input} ${Input%.*}_${Region}.fas -regions ${Region}
#
for i in *.fas; do awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" }END { printf "%s", n }'  ${i} > Temp.fas; mv Temp.fas ${i}; done
＃
for i in HPV51_Plus_FL-muscle_0001-0874.fas HPV51_Plus_FL-muscle_0874-2778.fas; do sed -E '/>/d' ${i} > ${i%.*}.seq; done
paste HPV51_Plus_FL-muscle_0874-7821.seq HPV51_Plus_FL-muscle_0001-0874.seq > HPV51_Plus_FL-muscle_Rearranged.seq
sed -i 's/\t//g' HPV51_Plus_FL-muscle_Rearranged.seq
sed -i 's/\-//g' HPV51_Plus_FL-muscle_Rearranged.seq
#
Seq=HPV51_Plus_FL-muscle.fa
# grep '>'  ${Seq} | sed 's/>//g' > IDs.txt; cp IDs.txt ${Seq%.*}.list
grep '>'  ${Seq} > ${Seq%.*}.list
paste HPV51_Plus_FL-muscle.list HPV51_Plus_FL-muscle_Rearranged.seq > HPV51_Plus_FL-muscle_Rearranged.fa
Seq=HPV51_Plus_FL-muscle_Rearranged.fa
sed 's/\t/\n/g' ${Seq} > ${Seq%.*}.fas
sed 's/\-//g' HPV51_Plus_FL-muscle_Rearranged.fas > HPV51_Plus_FL_Rearranged.fas
#
Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_Plus_FL_Rearranged.fas
SeqBase=`basename ${Seq}`; muscle -in ${Seq} -out ${SeqBase%.fas}-muscle.fa

Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_Plus_FL_Rearranged-muscle.fa
awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" }END { printf "%s", n }'  ${Seq} > Temp.fas; mv Temp.fas ${Seq}
ID=IDs.txt
sed -i 's/^/>/g' ${ID}
awk 'NR%2==0' ${Seq} | paste -d'\n' ${ID} - > ${Seq%.*}.fas
gblocks ${Seq%.*}.fas -t=d -e=-gb -b4=5 -d=y

847

cp ../1.fa.0 1.fa
#
function Nucleotide_NWK {
  TempName=`Second`
  Seq=$1
  cp ${Seq} ${Seq}.0
  #
  awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" }END { printf "%s", n }' ${Seq} > ${TempName}.fasta; mv ${TempName}.fasta ${Seq} # removal of line breaks in sequences
  grep '>' ${Seq} | sed 's/>//' | sed 's/\ /_/g' > ${Seq%.*}.list
  grep '>' ${Seq} | sed 's/>//' | awk '{print $1}' | awk -F "." '{print $1}'> ${Seq%.*}.shortlist
  sed 's/^/>/g' ${Seq%.*}.shortlist > ${TempName}.ID
  awk 'NR%2==0' ${Seq} | paste -d'\n' ${TempName}.ID - > ${TempName}.fasta; mv ${TempName}.fasta ${Seq} # replace the header in FASTA
  paste ${Seq%.*}.shortlist ${Seq%.*}.list > ${Seq%.*}.tbl
  muscle -in ${Seq} -out ${Seq%.*}-muscle.fa
  muscle -cluster neighborjoining -in ${Seq} -tree1 ${Seq%.*}.phylip -out ${Seq%.*}-muscle_phy.fa
  seqret -osformat2 phylip -sequence ${Seq%.*}-muscle.fa -outseq ${Seq%.*}-muscle.phy
  seqret -osformat2 phylip -sequence ${Seq%.*}-muscle_phy.fa -outseq ${Seq%.*}-muscle_phy.phy
  raxmlHPC -f a -# 20 -m GTRCAT -p 12345 -x 12345 -s ${Seq%.*}-muscle.phy -n ${Seq%.*}-muscle.phy.out
  raxmlHPC -f a -# 20 -m GTRCAT -p 12345 -x 12345 -s ${Seq%.*}-muscle_phy.phy -n ${Seq%.*}-muscle_phy_phy.out
}


gblocks ${Seq%.*}-muscle.fa -t=d -e=-gb -b4=5 -d=y
fasttree -gtr -nt ${Seq%.*}-muscle.fa-gb > ${Seq%.fas}-muscle.nwk

  cp ${Seq%.*}-muscle.fa ${TempName}.fa
  awk '!/^>/ { printf "%s", $0; n = "\n" } /^>/ { print n $0; n = "" }END { printf "%s", n }' ${Seq%.*}-muscle.fa > Temp.txt; mv Temp.txt ${Seq%.*}-muscle.fa
  awk 'NR%2==0' ${Seq%.*}-muscle.fa | paste -d'\n' ${Seq%.*}-muscle.shortlist - > Temp.txt; mv Temp.txt ${Seq%.*}-muscle.fa
  gblocks ${Seq%.*}-muscle.fa -t=d -e=-gb -b4=5 -d=y
  fasttree -gtr -nt ${Seq%.*}-muscle.fa-gb > ${Seq%.fas}-muscle.nwk

}


Input=HPV51_E6_DNA_Singled-muscle_phy.fas
seqret  -osformat2 phylip -sequence ${Input} -outseq ${Input%.*}.phy # convert MSA to phy

raxmlHPC -f a -# 20 -m GTRCAT -p 12345 -x 12345 -s ${Input} -n ${Input%.phy}.out

for i in *.nwk
do
  nw_display -s -S -w 1500 -v 20 -b 'opacity:0' -i 'font-size:8' -l 'font-family:serif;font-style:italic;font-size:20' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg
  inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf
done
#
for i in *.nwk; do nw_display -s -S -w 1500 -v 20 -b 'opacity:0' -i 'font-size:8' -l 'font-family:serif;font-style:italic;font-size:20' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
#
for i in *.nwk; do nw_display -s -S -w 1500 -v 20 -b 'opacity:0' -i 'font-size:8' -l 'font-family:serif;font-style:italic;font-size:12' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
for i in *.nwk; do nw_display -sr -S -w 1500 -v 20 -b 'opacity:0' -i 'font-size:8' -l 'font-family:serif;font-style:italic;font-size:12' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
for i in *.nwk; do nw_display -sr -S -w 2000 -v 20 -b 'opacity:0' -i 'font-size:8' -l 'font-family:serif;font-style:italic;font-size:12' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
#
for i in *.nwk; do nw_display -sr -S -w 2000 -v 20 -b 'opacity:0' -i 'font-size:6;fill:blue' -l 'font-family:serif;font-style:italic;font-size:10' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
#
for i in *.nwk; do nw_display -s -S -w 2000 -v 20 -b 'opacity:0' -i 'font-size:6;fill:blue' -l 'font-family:serif;font-style:italic;font-size:10' -d 'stroke-width:0.5;stroke:red' ${i} > ${i%.nwk}.svg; inkscape -f ${i%.nwk}.svg -A ${i%.nwk}.pdf; done
######################################
# Bookmarks import with pdftk
######################################
#
let j=0
for i in *.JPG; do let j+=1; mv $i "${j}.jpg"; done
rm Bookmarks.txt
touch Bookmarks.txt
echo '' > Bookmarks.txt
for i in HPV51*.pdf
do
  echo ${i%.pdf} >> Bookmarks.txt
done
sed -i 's/HPV51\_//g' Bookmarks.txt
#
pdftk HPV51*.pdf cat output All.pdf
#
touch Bookmarks.list
echo '' > Bookmarks.list
let j=0
for i in `cat Bookmarks.txt`
do
  let j+=1
  echo "BookmarkBegin" >> Bookmarks.list
  echo "BookmarkTitle: "${i} >> Bookmarks.list
  echo "BookmarkLevel: 1" >> Bookmarks.list
  echo "BookmarkPageNumber: " ${j} >> Bookmarks.list
done
#
Bookmark=Bookmarks.list; i=All.pdf; pdftk ${i} update_info ${Bookmark} output ${i%.pdf}_Bookmarked.pdf


awk 'FNR==NR{
  a[">"$1]=$2;next
}
$1 in a{
  sub(/>/,">"a[$1]"|",$1)
}1' names.txt seq.fa


awk 'FNR==NR{
  a[">"$1]=$2;next
}
$1 in a{
  sub(/>/,">"a[$1]"|",$1)
}1' 1-muscle.shortlist 1-muscle.fa > 2.fas

awk 'FNR==NR{
  a[">"$1]=$2;next
}
$1 in a{
  sub(/>/,">"a[$1]"",$1)
}1' 1-muscle.shortlist 1-muscle.fa > 3.fas

awk 'FNR==NR{a[">"$1]=$2;next} $1 in a{ sub(/>/,">"a[$1]"",$1)}1' 1-muscle.replace 4.fas

seqkit replace -p "(.+)" -r '{kv}|$1' -k 1-muscle.replace 1-muscle.fa > 4.fas

```
from ete3 import Tree
t = Tree( "((a,b),c);" )
t.show()

NWK = "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/20180721/Results/HPV51_E1_Trim-muscle_Trim.nwk"

from ete3 import Tree
NWK = "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/Temp/2_renamed.nwk"
t = Tree( NWK, format=1) # 输入Newick文件
t.show()

t.write(format=1) # 直接输出newick文件内容
#
from ete3 import Tree, TreeStyle
NWK = "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/20180721/Results/HPV51_E1_Trim-muscle_Trim.nwk"
t = Tree(NWK)
ts = TreeStyle()
ts.show_leaf_name = True
ts.mode = "c"
ts.scale=30
ts.arc_start = -90 # 0 degrees = 3 o'clock. 标注的数字以3点钟顺时针方向旋转角度
ts.arc_span = 270
t.show(tree_style=ts)
```

7z a HPV51_Plus_NCBI.7z HPV51_Plus.fasta  HPV51_Plus.genbank    HPV51_Plus.summary

cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/GB
grep '                     /country=' *.gb

grep 'country' *.gb | sed 's/\"//g' | sed "s/                     /\t/" | sed 's/\///' | sed 's/\.gb\://' | sed 's/country\=//'
grep 'country' *.gb | sed 's/\"//g' | sed "s/                     /\t/" | sed 's/\///' | sed 's/\.gb\://' | sed 's/country\=//' | awk '{print $1}' > Country.list # 含有地域信息的GenBank文件
grep 'country' *.gb | sed 's/\"//g' | sed "s/                     /\t/" | sed 's/\///' | sed 's/\.gb\://' | sed 's/country\=//' | awk '{print $1"/"$1"_"$2}' | sed 's/^/s\//' | sed 's/$/\/g/' > /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/GB/Country.sed

for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.1; done



Total 184 sequences
Among them, country (sampling location) information were found in 85 sequences (Country.list).
Among 99 sequences without sampling location information, two as below were artifical sequences (subcloning in E. coli)
  AB438954.gb:                     /lab_host="Escherichia coli JM109"
  AB438955.gb:                     /lab_host="Escherichia coli JM109"

70 sequences were ones from patents
grep '            PN   JP ' *.gb
grep '            PN   JP ' *.gb | wc -l # 70 sequences
grep '            PN   JP ' *.gb | awk -F "." '{print $1}' > Patent.list
for i in `cat Patent.list`; do mv -i  ${i}.gb ./Patent/; done

HPU45917,USA


grep '                     /isolate="Qv' *.gb 
grep '                     /isolate="Qv' *.gb | wc -l # 8 sequences

KF436866.gb:                     /isolate="Qv00885"
KF436867.gb:                     /isolate="Qv34180"
KF436868.gb:                     /isolate="Qv23639"
KF436870.gb:                     /isolate="Qv00891"
KF436875.gb:                     /isolate="Qv09427"
KF436882.gb:                     /isolate="Qv00495"
KF436883.gb:                     /isolate="Qv10565"
KF436884.gb:                     /isolate="Qv17234"

grep '                     /isolate="BF' *.gb 
grep '                     /isolate="BF' *.gb | wc -l # 6 sequences

KF436871.gb:                     /isolate="BF63"
KF436877.gb:                     /isolate="BF375"
KF436880.gb:                     /isolate="BF369"
KF436885.gb:                     /isolate="BF312"
KF436886.gb:                     /isolate="BF366"
KF436887.gb:                     /isolate="BF315"

grep -i '                     /isolate="rw' *.gb 
grep -i '                     /isolate="rw' *.gb | wc -l # 5 sequences

KF436869.gb:                     /isolate="Rw95"
KF436872.gb:                     /isolate="RW20"
KF436878.gb:                     /isolate="RW644"
KF436879.gb:                     /isolate="RW117"
KF436881.gb:                     /isolate="RW146"

grep '                     /note="sublineage ' *.gb

以下2序列来自1993文献

S40272.gb
S42990.gb

Totally, 85 + 25 = 110 sequences.

grep '                     /note="sublineage ' *.gb  | sed "s/                     /\t/" | sed 's/\///' | sed 's/\.gb\://' | sed 's/\"//g' | sed 's/note\=sublineage\ //'
grep '                     /note="sublineage ' *.gb  | sed "s/                     /\t/" | sed 's/\///' | sed 's/\.gb\://' | sed 's/\"//g' | sed 's/note\=sublineage\ //' | awk '{print $1"/"$1"_"$2}' | sed 's/^/s\//' | sed 's/$/\/g/'


      r-base-core:
          Depends: libc6 (>=2.27) but 2.26-0ubuntu2.1 is to be installed
          Depends: libcurl4 (>=7.28.0) but it is not installable
          Depends: libicu60 (>=60.1-1~) but it is not installable

ls RAxML_bestTree.*.out
for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.1; done
#
###################################################################
######## Replace the annotation in newick docuemnts #######
###################################################################
Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Sublineage.sed
for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.Sublineage; done
#
Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Country.sed
for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.Country; done
#
Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Sublineage_Underlined.sed
for i in *.Country; do sed -f ${Replace} < ${i} > ${i}.Sublineage; done
#
for i in RAxML_bestTree.*.Sublineage RAxML_bestTree.*.Country; do sed 's/\_\:/:/g' ${i} > ${i}.1; done
rename 's/RAxML_bestTree\.//' *.1; rename "s/.Country.Sublineage/.Final/" *.1
#
###################################################################
#
###################################################################
######## Replace the annotation in newick docuemnts #######
###################################################################
Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Sublineage.sed; for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.Sublineage; done; Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Country.sed; for i in RAxML_bestTree.*.out; do sed -f ${Replace} < ${i} > ${i}.Country; done; Replace=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Sublineage_Underlined.sed; for i in *.Country; do sed -f ${Replace} < ${i} > ${i}.Sublineage; done; for i in RAxML_bestTree.*.Sublineage RAxML_bestTree.*.Country; do sed 's/\_\:/:/g' ${i} > ${i}.1; done; rename 's/RAxML_bestTree\.//' *.1; rename "s/.Country.Sublineage/.Final/" *.1
#
7z l *.7z | awk '{print $5}' | grep '.1' | grep '.Final.1'
#
7z a *.7z *.1; mv -i *.Final.1 *.7z ../
#
###################################################################
#
cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/Temp
for Subject in M62877_??.fa; do Query=../HPV51_All_Small.fas; BLASTn ${Query} ${Subject}; done
#
for Query in "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_E1.fas  /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_L1.fas"; do Subject="/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_05_051_HPV51REF_M62877.fas"; BLASTn ${Query} ${Subject}; done

Query=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_L1.fas
Subject="/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_05_051_HPV51REF_M62877.fas"; BLASTn ${Query} ${Subject}
Query=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_E1.fas
Subject="/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_05_051_HPV51REF_M62877.fas"; BLASTn ${Query} ${Subject}


[绝命律师.Better.Call.Saul.S04E06.中英字幕.HDTVrip.720p-人人影视.mp4][5]

ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E01.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|590157519|0682c65184f8df697a90f38d7cee6c59|h=ezovbfnsp2e6u2tos5grtnzcn4sschdn|/
ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E02.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|557557341|e04605a6f17db8e5573d5e7d962a305b|h=fm43efgqbcozt732cdljyer2puln62wr|/
ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E03.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|498671846|8cb0cd2e061160d83eea4fdc4e3c11d7|h=ttqat6zbi6ad2mrdojfljobrswabqqkz|/
ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E04.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|559013951|93526a54a8f4f0d91c58599573db8e3e|h=nryvjngln4qdqg3jmn34u3fpocy3eb7g|/
ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E05.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|629277253|bca71f93dc5eaee5c366f2e25ddb1fe4|h=7w7yn55i3plprgpchifvgrdlgw6nkwzd|/
ed2k://|file|%E7%BB%9D%E5%91%BD%E5%BE%8B%E5%B8%88.Better.Call.Saul.S04E06.%E4%B8%AD%E8%8B%B1%E5%AD%97%E5%B9%95.HDTVrip.720p-%E4%BA%BA%E4%BA%BA%E5%BD%B1%E8%A7%86.mp4|569602860|9008cab5445c85e0f44ae09f7f3302a8|h=evrymnkymiwe6ozuoakk2mze2ew6ihne|/


HPV51_All.fasta,110,
HPV51_All_Small_E1.fas,15,
HPV51_All_Small.fas,46,
HPV51_All_Small_L1.fas,30,
HPV51_FL_20180914.fas,58,
HPV51_L2.fa,64,
HPV51_L2.fas,64,
HPV51_L2-muscle.fa,64,
HPV51_L2_Singled.fas,32,
HPV51_L2_Singled-muscle.fa,32,
HPV51_More.fas,0,需要重新制作
HPV51_Plus_L2L1.fas,35,
HPV51Ref.fas,1,Link

for i in *.fas; do echo $i ../$i; diff $i ../$i; echo '====================='; done | more


[HPV51_All_L2L1.fa](/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_L2L1.fa)
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_L2L1.fa

[HPV51_All_Small_L1.fas](/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_L1.fas)
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_L1.fas

cat /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/HPV51_All_Small_L1.fas /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_L2L1.fa > /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_L2L1_20180915.fa

Nucleotide_NWK /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_L2L1_20180915.fa

for i in *.fa*; do sed -i 's/\_cn//g' ${i}; sed -i 's/HPV0/NGS_/g' ${i}; done

Seq=1.fa
cp 1.list IDs.txt
awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' ${Seq} > ${Seq%.*}_Sub.fas

cp HPV51_All_Small_L1_6470-6890.list IDs.txt
Seq=HPV51_All_Small_L1.fas
awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' ${Seq} > ${Seq%.*}_Sub.fas
# HPV51_All_Small_L1_6470-6890.fas
mv HPV51_All_Small_L1_Sub.fas HPV51_All_Small_L1_6470-6890.fas -i

Region=6483-6903
Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180915/HPV51_All_Small_L1_6470-6890_Plus-muscle.fa
extractalign ${Input} ${Input%.*}_${Region}.fa -regions ${Region}

Region=1307-1928
Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_Small_E1_Plus-muscle.fa
extractalign ${Input} ${Input%.*}_${Region}.fa -regions ${Region}

Region=5531-7048
Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180915/HPV51_All_L1-muscle.fa
extractalign ${Input} ${Input%.*}_${Region}.fa -regions ${Region}

Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180915/HPV51_All_Small_L1_6470-6890_Plus-muscle.fa
bioawk -c fastx '{ print $name, length($seq) }' ${Seq}
bioawk -c fastx '{ print $name, length($seq) }' ${Seq} | sort -nk 2
bioawk -c fastx '{ print $name, length($seq) }' ${Seq} | awk '{print $2}' | uniq

rename 's/RAxML_bestTree\.//' *.1; rename "s/.Country.Sublineage/.Final/" *.1

7z l *.7z | awk '{print $5}' 
7z l *.7z | awk '{print $5}' | grep '.1'
7z l *.7z | awk '{print $5}' | grep '.1' | grep '.Final.1'

7z a *.7z *.1

mv -i *.Final.1 *.7z ../

Input=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_FL_20180914.fa
Extension="${Input##*.}"
Region=0099-0554
extractalign ${Input} ${Input%.*}_${Region}.${Extension} -regions ${Region}
Region=0562-0867
extractalign ${Input} ${Input%.*}_${Region}.${Extension} -regions ${Region}
Region=0876-2780
extractalign ${Input} ${Input%.*}_${Region}.${Extension} -regions ${Region}
Region=2722-3799
extractalign ${Input} ${Input%.*}_${Region}.${Extension} -regions ${Region}
#
cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180916223421
Nucleotide_NWK ../HPV51_All_FL_20180914_0099-0554.fa

cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180916223625
Nucleotide_NWK ../HPV51_All_FL_20180914_0562-0867.fa

cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180916223654
Nucleotide_NWK ../HPV51_All_FL_20180914_0876-2780.fa

cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/20180916223729
Nucleotide_NWK ../HPV51_All_FL_20180914_2722-3799.fa

Nucleotide_NWK HPV51_All_FL_20180914_0099-0554.fa
Nucleotide_NWK HPV51_All_FL_20180914_0562-0867.fa
Nucleotide_NWK HPV51_All_FL_20180914_0876-2780.fa
Nucleotide_NWK HPV51_All_FL_20180914_2722-3799.fa



HPV51_All_FL_20180914_0099-0554.fa
HPV51_All_FL_20180914_0562-0867.fa
HPV51_All_FL_20180914_0876-2780.fa
HPV51_All_FL_20180914_2722-3799.fa
#

for Input in HPV51_All_FL_20180914_0099-0554.fa HPV51_All_FL_20180914_0562-0867.fa HPV51_All_FL_20180914_0876-2780.fa HPV51_All_FL_20180914_2722-3799.fa; do Extension="${Input##*.}"; awk -f /media/ht/ht_5T/Work/References/Computer_Tips/Linux_Tips/Useful_Scripts/Work/dufa.awk -v L=${Input%.*}_`Second`.log ${Input} > ${Input%.*}_Trim.${Extension}; done

HPV51_All_FL_20180914_0099-0554_Trim.fa
HPV51_All_FL_20180914_0562-0867_Trim.fa
HPV51_All_FL_20180914_0876-2780_Trim.fa
HPV51_All_FL_20180914_2722-3799_Trim.fa

Nucleotide_NWK ../HPV51_All_FL_20180914_0099-0554_Trim.fa
Nucleotide_NWK ../HPV51_All_FL_20180914_0562-0867_Trim.fa
Nucleotide_NWK ../HPV51_All_FL_20180914_0876-2780_Trim.fa
Nucleotide_NWK ../HPV51_All_FL_20180914_2722-3799_Trim.fa

HPV51_All_Small_L1_6470-6890_Plus-muscle_phy_phy.out.Final.1

/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_Small_L1_6470-6890_Plus-muscle_phy_phy.out.Final.1

library(tidyverse); library(ggtree)
Tree <- read.tree("/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV51_Detect/Seq/MSA/HPV51_All_Small_L1_6470-6890_Plus-muscle_phy_phy.out.Final.1")
ggtree(Tree) + geom_tiplab()

HPV51_All_FL_20180914_0099-0554_Trim-muscle.phy.out.Final.1
HPV51_All_FL_20180914_0099-0554_Trim-muscle_phy_phy.out.Final.1
HPV51_All_FL_20180914_0562-0867_Trim-muscle.phy.out.Final.1
HPV51_All_FL_20180914_0562-0867_Trim-muscle_phy_phy.out.Final.1
HPV51_All_FL_20180914_0876-2780_Trim-muscle.phy.out.Final.1
HPV51_All_FL_20180914_0876-2780_Trim-muscle_phy_phy.out.Final.1
HPV51_All_FL_20180914_2722-3799_Trim-muscle.phy.out.Final.1
HPV51_All_FL_20180914_2722-3799_Trim-muscle_phy_phy.out.Final.1
HPV51_All_FL_20180914_Rearrange_Trim-muscle.phy.out.Final.1
HPV51_All_FL_20180914_Rearrange_Trim-muscle_phy_phy.out.Final.1


HPV51_All_FL_20180914_0876-2780_Trim-muscle.phy.out.Final.svg
HPV51_All_FL_20180914_2722-3799_Trim-muscle.phy.out.Final.svg
HPV51_All_FL_20180914_Rearrange_Trim-muscle.phy.out.Final.svg
HPV51_All_FL_20180914_Trim-muscle.phy.out.Final.svg
HPV51_All_L1_5531-7048_Trim-muscle.phy.out.Final.svg
