/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-J_0546F.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-J_3806R.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-J_W1FT0119009.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-J_W1RT0119010.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-K_3262F.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-K_6747R.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-K_W1FT0119011.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-K_W1RT0119012.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-K_W2FT0121016.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-L_0852R.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-L_6262F.ab1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01-L_W1FT0119013.ab1


1 fsta scf pdf jpg qual seq txt

ls -1 | rev | awk -F '.' '{print $1}' | sort | uniq | rev
1
ab1
fsta
scf
pdf
jpg
qual
seq
txt

for i in 1 fsta scf pdf jpg qual seq txt; do echo ${i}; ls -1 *.${i} | wc -l; echo '==================='; done

/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas

SangerMap /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas 01.fq

Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
Seq=`basename $PWD`.fq
Seq_Base=`basename ${Seq}`
Ref_Base=`basename ${Ref%.*}`
BAM=${Seq_Base%.*}_vs_`basename ${Ref_Base}`.bam
magicblast -query ${Seq} -subject ${Ref} -infmt fastq -outfmt sam | samtools view -bS -F 4 -> ${Seq_Base%.*}_vs_`basename ${Ref_Base}`_Mapped.bam # ÐòÁÐÖÐ¾ø¶Ô½ûÖ¹`-`£¬ÕâÀà·ûºÅÊÇ¾­¹ýMSA²úÉúµÄ
BAM=${Seq_Base%.*}_vs_`basename ${Ref_Base}`_Mapped.bam
samtools sort $BAM > "${BAM%.bam}.sort.bam"
samtools index "${BAM%.bam}.sort.bam"
samtools view -h -F 4 "${BAM%.bam}.sort.bam" > "${BAM%.bam}.sam"
setsid tablet "${BAM%.bam}.sort.bam" ${Ref}

Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
BAM=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/01/01_vs_A07039_HPV39REF_M62849_Mapped.sort.bam
samtools mpileup -gf ${Ref} ${BAM} | bcftools view | vcfutils.pl varFilter | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}'

samtools mpileup -gf ${Ref} ${BAM} > 1.bcf


cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/
find ./ -type f -name "*.bam" | xargs -t mv ./


* [text processing - Delete lines containing empty fields - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/347843/delete-lines-containing-empty-fields)
* [Extracting Columns From Vcf File Using Vcftools/Perl](https://www.biostars.org/p/51076/)
* [converting VCF to CSV](https://www.biostars.org/p/153238/)

samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv
#

for Seq in {01..15}; do SangerMap ${Ref} ${Seq}.fq; done
for BAM in *.sort.bam; do samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv; done
rename "s/A07039_HPV39REF_M62849/HPV39Ref/" *.csv *.bam *.bai *.sam
#
ls -I "*.fq" | xargs rm
#
Ref=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39Rev.fas
for Seq in {01..15}; do SangerMap ${Ref} ${Seq}.fq; done
rename "s/A07039_HPV39REF_M62849/HPV39Ref/" *.csv *.bam *.bai *.sam

for BAM in *_HPV39Ref_Mapped.sort.bam; do samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv; done


Full length, 7833 bp
5925

#####################################################################
for i in {01..15}; do mkdir ${i}; cp ${i}-*.ab1 ./${i}; done
#
Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
for Dir in {01..15}; do cd ${Dir}; ab2fq; SangerMap ${Ref} ${Dir}.fq; cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1/; done
find . -type f -name "*.fq" | xargs -i cp {} ./
#
cd /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1
Ref=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_6001_7800_Origin.fas
for Seq in {01..15}; do SangerMap ${Ref} ${Seq}.fq; done
samtools faidx ${Ref}
for BAM in *.sort.bam; do samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv; done
Target=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180918214958
mv *.csv *.bam *.bai *.sam ${Target}
for i in {01..15}; do echo ${i}; done | xargs mv ${Target}
#
rename "s/vs_HPV39_6001_7800_Origin_Mapped/2/" *.*
rename "s/_vs_A07039_HPV39REF_M62849_Mapped//" *.*
#
Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
for BAM in ??.sort.bam; do samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv; done

Ref=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_6001_7800_Origin.fas
for BAM in ??_2.sort.bam; do samtools mpileup -gf ${Ref} ${BAM} | bcftools view | awk 'BEGIN {OFS ="," ; FS = "\t"};{print $2 "\t" $4 "\t" $5}' | sed '/^\t/d' | sed 's/<\*>//' | sed 's/\,//' | awk '$3!=""' > ${BAM%.sort.bam}.csv; done
