awk 'BEGIN { OFS = "," } {print $0, "z"}' input.csv 

awk 'BEGIN { OFS = "," } {print '"${i}"' "\t" $0}' 14.csv

for i in ??.csv; do awk 'BEGIN { OFS = "," } {print '"${i%.csv}"' "\t" $0}' ${i} > ${i}.1; done
for i in ??_2.csv; do awk 'BEGIN { OFS = "," } {print '"${i%_2.csv}"' "\t" $0}' ${i} > ${i}.1; done
for i in *.1; do sed -i '/ALT$/d' ${i}; done

```{r, eval = FALSE, echo = FALSE}
# 合并BAM输出VCF形成的CSV的2个表
CSV2 <- list.files()[grep("_2.csv", list.files())]
CSV <- list.files()[grep(".csv", list.files())]
CSV1 <- CSV[-grep("_2.csv", CSV)]
#
for ( i in 1 : 15 ) {
    x <- read.csv(CSV2[i], sep = "\t")
    x[,1] <- x[,1] + 5925
    x[(x[,1] > 7833),1] <- x[,1][x[,1] > 7833] -7833
    # x <- x[order(x[,1]),]
    x <- rbind(x, read.csv(CSV1[i], sep = "\t"))
    x <- unique(x)
    x <- cbind(gsub(".csv", "", CSV1[i]), x)
    names(x)[1] <- "FQ"
    Input <- paste0("../", CSV1[i], ".2")
    write.csv(x, Input, row.names = FALSE)
}
```

Biostrings

biocLite("sangerseqR")

biocLite("Biostrings")

library(Biostrings)

/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/AB1
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921

```{r, eval = FALSE, echo = FALSE}
setwd("/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921")
library(seqinr); library(readxl); library(stringr)
Input <- "/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas"
x <- read.fasta(Input)
Seq <- x[[1]]
Excel <- "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180918/CSV.xlsx"
Table <- read_excel(Excel, sheet = "1")
Table <- data.frame(Table[,1:5])
Table[Table[,5] == "Y", ]
#
Table <- data.frame(read_excel(Excel, sheet = "2"))
Table <- subset(Table, Table[,5] == "Y")
#
OriginalSeq <- Seq
for ( Number in 1 : 15 ) {
    Seq <- OriginalSeq
    Table_Change <- subset(Table, Table[,1] == Number)
    Seq[Table_Change[,2]] <- Table_Change[,4]
    write.fasta(names = paste0("Sanger_", str_pad(1:15, 2, pad = "0")[Number]),
                                                        sequences=Seq,
                                                        file.out = paste0(str_pad(1:15, 2, pad = "0")[Number], ".fas"))
}
```


write.fasta(names="DEN-1", sequences=dengueseq, file.out="den1.fasta")

/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/01.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/02.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/03.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/04.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/05.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/06.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/07.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/08.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/09.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/10.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/11.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/12.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/13.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/14.fas
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/Sanger/Results/20180921/15.fas


Input=01.fas
#
cp ../??.fas ./
#
htfunction
for Input in *.fas; do
echo `grep '>' ${Input}`_2001-7833-0001 > ${Input%.*}.fasta
Region="2001-8000"
extractalign ${Input} ${Input%.*}_${Region}.fa -regions ${Region}
sed  '/^>/d' ${Input%.*}_${Region}.fa >> ${Input%.*}.fasta
rm ${Input%.*}_${Region}.fa
Region="0001-2000"
extractalign ${Input} ${Input%.*}_${Region}.fa -regions ${Region}
sed  '/^>/d' ${Input%.*}_${Region}.fa >> ${Input%.*}.fasta
rm ${Input%.*}_${Region}.fa
RemoveFastaBreak ${Input%.*}.fasta
rm ${Input%.*}.fasta
mv ${Input%.*}_NoBreak.fasta ${Input%.*}_2.fas
done
