Query=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
htfunction
BLASTn ${Query} ${Subject}
#
[HPV39_100_Patent_Free.fas](/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/A7/20180906/HPV39_100_Patent_Free.fas)
#
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/A7/20180906/HPV39_100_Patent_Free.fas # 86 sequences
#
cat HPV39_vs_Papillomaviridae_20180604.csv | awk '{print $2}' | sort | uniq | wc -l # 979 sequences
#
cat HPV39_vs_Papillomaviridae_20180604.csv | awk '{print $2}' | sort | uniq | awk -F "." '{print $1}'
#
Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/A7/20180906/HPV39_100_Patent_Free.fas
cat ${Seq} | grep '>' | sed 's/>//' | awk -F "." '{print $1}'
 | sort | uniq | awk -F "." '{print $1}'
#
Seq=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_vs_Papillomaviridae_20180604.csv
cat ${Seq} | awk '{print $2}' | sort | uniq | awk -F "." '{print $1}'
#
grep -Fvf /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/A7/20180906/HPV39_100_Patent_Free.ID /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_vs_PV.ID
#
awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' all.fas > subset.fas
#
awk 'BEGIN{while((getline < '"${ID}"')>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' all.fas > subset.fas

grep -Fvf \
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/A7/20180906/HPV39_100_Patent_Free.list \
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_vs_PV.list \
#
cp HPV39_Possible.list IDs.txt
awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f'  1.fas > 2.fas
#
Query=/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_Possible.fas
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20170326/A07039_HPV39REF_M62849.fas
BLASTn ${Query} ${Subject}

```
HPV39_Possible_vs_HPV39Ref.csv
HPV39Ref_vs_HPV39_Possible.csv
awk '!/^>/ { next } { getline seq } length(seq) >= 100 { print $0 "\n" seq }' HPV39_Possible.fas > HPV39_Possible_100.fas # 885 sequences longer than 100 bp extracted from 893 sequences
#
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_Possible_vs_HPV39Ref.csv
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39Ref_vs_HPV39_Possible.csv
#
# [blastn_Treat.R](/media/ht/ht_5T/Work/References/Computer_Tips/Linux_Tips/Useful_Scripts/Work/blastn_Treat.R)
#
CSV <- c("/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_Possible_vs_HPV39Ref.csv", "/media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39Ref_vs_HPV39_Possible.csv")
#
LQ666453.1 LQ666442.1 LQ666431.1 LQ666422.1 HQ834578.1 HQ834577.1 GQ161543.1 EU911879.1 EU911877.1 EU911875.1 EU911853.1 EU911739.1 EU911488.1 EU911269.1 EU911268.1 EU911267.1 EU911266.1 EU911265.1 EU911264.1 EU911263.1 EU911262.1 EU911246.1 EU911245.1 EU911244.1 EU911243.1 EU911242.1 EU911241.1 EU911240.1 EU911239.1 EU911238.1 EU911237.1 EU911121.1 EU911120.1 EU911119.1 EU911118.1 EU911117.1 EU911116.1 EU911045.1 EU911044.1 EU911043.1 EU911042.1 EU911041.1 EU911040.1 EU911039.1 HZ161036.1 HZ161027.1 HZ160969.1 HZ161047.1 LN609275.1 JC489888.1 JB742017.1 JB223517.1 HV708330.1 HV708319.1 HV708310.1 HV708252.1 EF422221.1 HD053459.1 HD053448.1 HD053439.1 HD053380.1 HC889023.1 HC888996.1 CS645799.1 AX838909.1
```

Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
cp HPV39_Possible_Selected.list IDs.txt # /media/ht/ht_5T/Work/Projects/CC_n_HPV/Alpha_HPV_Project/HPV39_Detect/Seq/HPV39_Possible_Selected.list
#
awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' ${Ref} > HPV39_Possible_Selected.fas # 65 sequences from PV DB longer than 100 bp and identity with HPV39Ref more than 90%

/media/ht/ht_5T/Work/Projects/CC_n_HPV/Integration/
/media/ht/ht_5T/Work/Projects/CC_n_HPV/Integration/301_Samples/Analysis/magicblast_Results/Extracted_BAM/

/media/ht/ht_5T/Work/Projects/CC_n_HPV/Integration/301_Samples/Analysis/magicblast_Results/Extracted_BAM/8K/39

#######################################

* [Consensus FASTA from BAM files - SEQanswers](http://seqanswers.com/forums/showthread.php?t=11536)
* [Why MAQ consensus seq better than SAMtools consensus ?? - SEQanswers](http://seqanswers.com/forums/showthread.php?t=8910)
* [How to get the consensus sequence from a BAM alignment](https://www.biostars.org/p/186536/)