## PV248 vs Papillomaviridae

```
Query=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV_Ref_FL_248.fas
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
BLASTn ${Query} ${Subject}
#
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
OriginalSubject=${Subject}
for Query in /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/*.fas; do Subject=${OriginalSubject}; BLASTn ${Query} ${Subject}; done
#
cd /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/20180925131120
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
OriginalSubject=${Subject}
for Query in `cat FAS.List`; do Subject=${OriginalSubject}; BLASTn ${Query} ${Subject}; done
#
Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
Query=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/FAS/PV01_09_035_HPV35REF_X74477.fas
BLASTn ${Query} ${Subject}
#
Input <- "Papillomaviridae_20180604_NoBreak_0100_vs_PV_Ref_FL_248.csv"
x <- read.csv(Input, sep = "\t", header = FALSE)
names(x)  <-  c("Seq", "PV248", "Identity", "Align_Len", "mismatches", "gap", "Q_start", "Q_end", "db_start", "db_end", "bit", "Score") 
y  <-  subset(x, Identity >= 90)
y  <-  subset(y, Align_Len >= 100)
HPV <- unique(y[,2])
for ( Genotype in HPV ) {
	x <- subset( y, y[,2] == Genotype )
	Input <- paste0(Genotype, ".ID")
	write.table( x[,1], Input, col.names = FALSE, row.names = FALSE)
}
#
for i in *.ID; do sed -i 's/\"//g' ${i}; done
#
for i in *.ID; do cat ${i} | sed 's/\"//g'  | awk -F '.' '{print $1}' > ${i}.1; done
#
ID=HPV39_All_20180923_L1_Partial.ID
#
Seq=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Papillomaviridae_20180604_NoBreak_0100.fas
for ID in *.ID; do cp ${ID} IDs.txt; awk 'BEGIN{while((getline<"IDs.txt")>0)l[">"$1]=1}/^>/{f=l[$1]?1:0}f' ${Seq} > ${ID%.ID}.fas; done
#
for i in *.fas; do FasSimple ${i}; done
rename "s/.Short//" *.*
7z a ../Short.7z *.fas
ls -1 *.fas > Short.List
for i in `cat Short.List`; do Nucleotide_NWK ${i}; done
```

## [Updating R on Ubuntu | R-bloggers](https://www.r-bloggers.com/updating-r-on-ubuntu/)

## [ggtree installation · GuangchuangYu/ggtree Wiki · GitHub](https://github.com/GuangchuangYu/ggtree/wiki/ggtree-installation)

```
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")
# biocLite("BiocUpgrade") # you may need this
biocLite("ggtree")
#
sudo apt-get install libfftw3-3 libfftw3-dev libtiff5-dev
```

```
# rsync -au Source Destination
# Source -> Destination backup with update
rsync -au /media/ht/ht_5T/Stored_Softs/VirtualBox /media/ht/ht_6T_1/Stored_Softs/VirtualBox

Destination=/media/ht/ht_2T/Movies
From=/media/ht/Seagate_1T/Movies
rsync -au ${From} ${Destination}
```

## [shell - find the max value of column 1 and print respective record from column 2 from file - Unix & Linux Stack Exchange](https://unix.stackexchange.com/questions/170204/find-the-max-value-of-column-1-and-print-respective-record-from-column-2-from-fi)

awk -v max=0 '{if($1>max){want=$2; max=$1}}END{print want} ' version.log
sort -nrk1,1 filename | head -1 | cut -d ' ' -f3

- [MAFFT - a multiple sequence alignment program](https://mafft.cbrc.jp/alignment/software/)

```
Seq=PV01_09_016_HPV16REF_NC_001526.fas
Extension="${Seq##*.}"; mafft ${Seq} > ${Seq%.${Seq##*.}}-mafft.${Extension}
#
Seq=PV01_07_018_HPV18REF_NC_001357.fas
Extension="${Seq##*.}"; mafft ${Seq} > ${Seq%.${Seq##*.}}-mafft.${Extension}
#
Seq=PV01_09_052_HPV52REF_X74481.fas
Extension="${Seq##*.}"; mafft ${Seq} > ${Seq%.${Seq##*.}}-mafft.${Extension}
#
Seq=PV01_09_058_HPV58REF_D90400.fas
Extension="${Seq##*.}"; mafft ${Seq} > ${Seq%.${Seq##*.}}-mafft.${Extension}

cd /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/FAS
PV01_10_011_HPV11REF_M14119

for i in `cat Short.List`; do Mafft ${i}; done


	If unsure which option to use, try 'mafft --auto input > output'.
	For more information, see 'mafft --help', 'mafft --man' and the mafft page.

	The default gap scoring scheme has been changed in version 7.110 (2013 Oct).
	It tends to insert more gaps into gap-rich regions than previous versions.
	To disable this change, add the --leavegappyregion option.

THIS MANUAL IS FOR V6.2XX (2007)
       Recent versions (v7.1xx; 2013 Jan.) have more features than those
       described below.  See also the tips page at
       http://mafft.cbrc.jp/alignment/software/tips0.html

cd /media/ht/ht_5T/Work/References/eBooks/Statistics

cd /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/FAS

   0 PV22_01_001_CPV5_FJ492742-mafft.fas
   0 PV11_03_003_CPV6_FJ492743-mafft.fas
   0 PV11_01_001_UuPV1_DQ180494-mafft.fas
   0 PV11_01_001_PcPV1_AY904723-mafft.fas
   0 PV11_01_001_LrPV1_AY904722-mafft.fas
   0 PV04_03_003_OaPV2_U83595-mafft.fas
   0 PV03_XX_mSD2_HPV-mSD2meta_KC113191-mafft.fas
   0 PV03_XX_mFS1_HPV-mFS1meta_JF966373-mafft.fas
   0 PV03_XX_mFD2_HPV-mFD2meta_JF966376-mafft.fas
   0 PV03_XX_mCG3_HPV-mCG3meta_JF966379-mafft.fas
   0 PV03_XX_162_HPV162REF_JX413108-mafft.fas
   0 PV03_12_148_HPV148REF_GU129016-mafft.fas
   0 PV03_07_155_HPV155REF_JF906559-mafft.fas
   0 PV03_07_149_HPV149REF_GU117629-mafft.fas
   0 PV03_01_065_HPV65REF_X70829-mafft.fas
   0 PV02_01_047_HPV47REF_M32305-mafft.fas
   0 PV02_01_001_MfPV1_EF028290-mafft.fas
   0 PV01_14_106_HPV106REF_DQ080082-mafft.fas
   0 PV01_09_016_HPV16REF_NC_001526-mafft.fas


https://mafft.cbrc.jp/alignment/software/mafft_7.407-1_amd64.deb

1 ./PV03_01_065_HPV65REF_X70829.fas
1 ./PV03_07_155_HPV155REF_JF906559.fas
1 ./PV03_12_148_HPV148REF_GU129016.fas
1 ./PV03_XX_162_HPV162REF_JX413108.fas
1 ./PV03_XX_mFD2_HPV-mFD2meta_JF966376.fas
1 ./PV03_XX_mFS1_HPV-mFS1meta_JF966373.fas
1 ./PV11_01_001_PcPV1_AY904723.fas
1 ./PV11_01_001_UuPV1_DQ180494.fas
1 ./PV11_03_003_CPV6_FJ492743.fas
1 ./PV02_01_047_HPV47REF_M32305.fas
1 ./PV01_14_106_HPV106REF_DQ080082.fas
1 ./PV02_01_001_MfPV1_EF028290.fas
1 ./PV03_07_149_HPV149REF_GU117629.fas
1 ./PV03_XX_mCG3_HPV-mCG3meta_JF966379.fas
1 ./PV03_XX_mSD2_HPV-mSD2meta_KC113191.fas
1 ./PV04_03_003_OaPV2_U83595.fas
1 ./PV11_01_001_LrPV1_AY904722.fas
1 ./PV22_01_001_CPV5_FJ492742.fas


* [OS X - Reset TeamViewer ID Issue : teamviewer](https://www.reddit.com/r/teamviewer/comments/5q2o7d/os_x_reset_teamviewer_id_issue/)
* [Crack Teamviewer 12, reset Teamviewer ID (also for mac OS X) - YouTube](https://www.youtube.com/watch?v=3Dn5LY9ouBU&feature=youtu.be)
* [[Linux] How to Reset TeamViewer ID on Linux @ 香腸炒章魚 :: 痞客邦 ::](http://mitblog.pixnet.net/blog/post/44051655-%5Blinux%5D-how-to-reset-teamviewer-id-on-linux)
* [13.10 - How to change Teamviewer ID after cloning - Ask Ubuntu](https://askubuntu.com/questions/423314/how-to-change-teamviewer-id-after-cloning)
* [How to Reset TeamViewer ID – TheITBros](https://theitbros.com/reset-teamviewer-id/)
* [MAFFT - a multiple sequence alignment program](https://mafft.cbrc.jp/alignment/software/)

## Backup

```
# rsync -au Source Destination
# Source -> Destination backup with update
rsync -au /media/ht/ht_5T/Stored_Softs/VirtualBox /media/ht/ht_6T_1/Stored_Softs/VirtualBox

Destination=/media/ht/ht_2T/Movies
From=/media/ht/Seagate_1T/Movies
rsync -au ${From} ${Destination}
```