## Backup

```
From=/media/ht/Seagate_1T/Movies
Destination=/media/ht/18T/Movies/Movies
rsync -au ${From} ${Destination}
#
From=/media/ht/Seagate_1T/Movies
Destination=/media/ht/18T/Movies
rsync -au ${From} ${Destination}
#
From=/media/ht/Seagate_1T/Movies
Destination=/media/ht/ht_2T/Movies/Movies
rsync -au ${From} ${Destination}
```

## [How to fix "The system is running in low-graphics mode" error?](https://askubuntu.com/questions/141606/how-to-fix-the-system-is-running-in-low-graphics-mode-error)

## CPU/GPU listup

* [command line - How to get the GPU info?](https://askubuntu.com/questions/5417/how-to-get-the-gpu-info)

```
cat /proc/cpuinfo # for CPU information
lspci
lspci | grep VGA # Show Graphic card information
lshw
```

## Alignment of HPV16 sequence collection

```
rm sources.list hosts grub.cfg lubuntu-rc.xml* .bashrc

#
cd /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/HPV16
Seq=caHPV16.fas # duplicated from /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/HPV16/caHPV16.fas, 10328 sequences
Number=7000
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) >= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}_${Number}.${Extension} # /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/20180925/HPV16/caHPV16_7000.fas, 3668 sequences
#
Seq=caHPV16.fas
Number=1000
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) <= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}_${Number}.${Extension} # caHPV16_0100-1000.fas, 5058 sequences
###############################
# 1001-2000 bp
###############################
Seq=caHPV16.fas
Number=1001
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) >= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}_${Number}.${Extension}
Seq=${Seq%.*}_${Number}.${Extension}
Number=2000
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) <= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}-${Number}.${Extension} # caHPV16_1001_2000.fas, 1519 sequences
###############################
# 2001-7000 bp
###############################
Seq=caHPV16.fas
Number=2001
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) >= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}_${Number}.${Extension}
Seq=${Seq%.*}_${Number}.${Extension}
Number=7000
Extension="${Seq##*.}"; awk '!/^>/ { next } { getline seq } length(seq) <= '"${Number}"' { print $0 "\n" seq }' ${Seq} > ${Seq%.*}-${Number}.${Extension} # caHPV16_1001_2000.fas, 1519 sequences
#
for i in *_*-*.fas; do RemoveDup ${i}; done
ls *_*-*.fas
# caHPV16_0100-1000.fas
    Input sequences: 5058
    Output sequences: 1707
    Reduction: 66.25%
# caHPV16_1001-2000.fas
    Input sequences: 1519
    Output sequences: 1051
    Reduction: 30.81%
# caHPV16_2001-7000.fas
    Input sequences: 83
    Output sequences: 82
    Reduction: 1.20%
for i in *_Trim.fas; do cat ${i} /media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_09_016_HPV16REF_NC_001526.fas > ${i}.1; done
for i in *.1; do sed -i "s/PV01_09_016_HPV16REF_NC_001526/HPV39Ref/g" ${i}; done
#
for i in caHPV16_0100-1000_Trim.fas caHPV16_2001-7000_Trim.fas caHPV16_1001-2000_Trim.fas; do Nucleotide_NWK ${i}; done
#
for i in caHPV16_0100-1000_Trim.fas caHPV16_2001-7000_Trim.fas caHPV16_1001-2000_Trim.fas; do mafft ${i} > ${i%.*}-mafft.fas; done
#
for Query in caHPV16_0100-1000_Trim.fas caHPV16_2001-7000_Trim.fas caHPV16_1001-2000_Trim.fas; do Subject=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_09_016_HPV16REF_NC_001526.fas; BLASTn ${Query} ${Subject}; done
#
for i in caHPV16_0100-1000.log  caHPV16_1001-2000.log  caHPV16_2001-7000.log; do awk '$3!=""' ${i} > ${i}.1; done
#
Ref=/media/ht/ht_5T/Work/Bioinformatics/DNA_DB/Virus/Papillomaviridae/Important_Seq/PV248_Divided/PV01_09_016_HPV16REF_NC_001526.fas

#
# Backup
mv caHPV16_7000.fas caHPV16_7000.fas.0
mv caHPV16_7000.fas.Short caHPV16_7000.fas


## Biopyphon

- [python - Biopython Phylogenetic Tree Edit labels in SVG file - Stack Overflow](https://stackoverflow.com/questions/33147651/biopython-phylogenetic-tree-edit-labels-in-svg-file)
- [pylab.savefig Python Example](https://www.programcreek.com/python/example/2475/pylab.savefig)
- [Topic: converts](https://github.com/topics/converts?o=desc&s=stars)
- [Inconsistent image size with savefig(...,type='png') · Issue #8542 · matplotlib/matplotlib](https://github.com/matplotlib/matplotlib/issues/8542/)

```

pip3 install biopython

#python3

from Bio import Phylo
import pylab

def get_label(leaf):
    return leaf.name

f = 'path/to/my/file'
tree = Phylo.read(f, 'newick')
tree.ladderize()
Phylo.draw(tree, label_func=get_label, do_show=False)
pylab.axis('off')
pylab.savefig('tree2.svg',format='svg', bbox_inches='tight', dpi=300)

tree = Phylo.read('example.dnd', 'newick')
tree = Phylo.read('HPV51_All_E1_16.nwk', 'newick')
print tree


teamviewer --daemon start
teamviewer --ziplog


ht-Vision-Ubuntuliu0330

sudo ifconfig enp3s0 down
sudo ifconfig enp3s0 hw ether 00:11:22:33:44:55
sudo ifconfig enp3s0 up

sudo dpkg -i /media/ht/ht_5T/Stored_Softs/Cross_Platform_Softs/TeamViewer/13/teamviewer_13.2.13582_amd64.deb

----- BEGIN LICENSE -----
sgbteam
Single User License
EA7E-1153259
8891CBB9 F1513E4F 1A3405C1 A865D53F
115F202E 7B91AB2D 0D2A40ED 352B269B
76E84F0B CD69BFC7 59F2DFEF E267328F
215652A3 E88F9D8F 4C38E3BA 5B2DAAE4
969624E7 DC9CD4D5 717FB40C 1B9738CF
20B3C4F1 E917B5B3 87C38D9C ACCE7DD8
5F7EF854 86B9743C FADC04AA FB0DA5C0
F913BE58 42FEA319 F954EFDD AE881E0B
------ END LICENSE ------

